### PR TITLE
[Repo Command] Handle add repo mkpath exception

### PR DESCRIPTION
**Pull request by sguillope**
Hi, this is my first time at coding in Ruby but wanted to give it a shot. I apologize if this is completely off.

The issue description didn't specify what to do after handling the exception so I assumed it needed to be raised as an `Informative` (not sure what else could be done).
I've also taken the liberty to do a bit of refactoring to make the main `run` method more readable.

Comments welcomed!
Thanks

### CODE REVIEW

File: lib/cocoapods/command/repo/add.rb [hunk 1]
```diff
@@ -39,16 +39,48 @@ def validate!
...
+        # `config.repos_dir`.
+        #
+        # @return [void]
+        #
+        # @raise  If the directory cannot be created due to a system error.
+        #
+        def create_repos_dir
+          config.repos_dir.mkpath
+        rescue SystemCallError => e
+          raise Informative, "Could not create repos directory '#{config.repos_dir}'.\n#{e.class.name}: #{e.message}"
```
Comment (sguillope) [inline lib/cocoapods/command/repo/add.rb]: The real change is here. The rest of the changes is only refactoring.

File: lib/cocoapods/command/repo/add.rb [hunk 2]
```diff
@@ -39,16 +39,48 @@ def validate!
...
+        end
+
+        # Clones the git spec-repo according to parameters passed to the
+        # command.
+        #
+        # @return [void]
+        #
+        def clone_repo
+          Dir.chdir(config.repos_dir) do
```
Comment (sguillope) [inline lib/cocoapods/command/repo/add.rb]: Should we handle a potential `SystemCallError` here as well and in `checkout_branch`?

File: lib/cocoapods/command/repo/add.rb (hunk 1 above)
Comment (orta) [inline lib/cocoapods/command/repo/add.rb]: Could this say `"Could not create '#{config.repos_dir}', the CocoaPods repo cache directory.\n#{e.class.name}: #{e.message} ` - I think it's a bit clearer about why that folder has to exist.

Comment (sguillope) [inline lib/cocoapods/command/repo/add.rb]: :thumbsup:
FYI this is an example of how the output would look like:

```
% pod setup
Setting up CocoaPods master repo
[!] Could not create '/Users/sguillope/.cocoapods/repos', the CocoaPods repo cache directory.
Errno::EACCES: Permission denied - /Users/sguillope/.cocoapods/repos
```

File: lib/cocoapods/command/repo/add.rb [hunk 3]
```diff
@@ -39,16 +39,49 @@ def validate!
...
+        # Creates the repos directory specified in the configuration by
+        # `config.repos_dir`.
+        #
+        # @return [void]
+        #
+        # @raise  If the directory cannot be created due to a system error.
+        #
+        def create_repos_dir
+          config.repos_dir.mkpath
+        rescue SystemCallError => e
```
Comment (segiddins) [inline lib/cocoapods/command/repo/add.rb]: I'm ok with just `rescue => e` here

**Comment by segiddins** [comment]
:+1: other than that one comment. thanks for the great PR, @sguillope !

**Comment by segiddins** [comment]
Will need a quick rebase due to the CHANGELOG comment, then I'll merge. :rocket:

**Comment by segiddins** [comment]
<3

**Pull request merged**